src/panel.js

```diff
@@ -117,24 +117,56 @@ const addSpan = (wrapper, stringId) => {
   setClassAndAppend(wrapper, span);
 };
 
-const getActiveHostname = async() => {
+const getRootDomain = (hostname) => {
+  let parts = hostname.split(".");
+
+  // Default array length of hostname split
+  let defaultDomainLength = 3;
+
+  // If there's no subdomain, return hostname as is.
+  if (parts.length < defaultDomainLength) {
+    return hostname;
+  }
+
+  let lastArrayLength = parts[(parts.length - 1)].length;
+
+
+  if (lastArrayLength < 3) {
+    // Returns true if last part of hostname is two-characters, meaning it's a
+    // country-level domain.
+    let nextToLastItem = parts[(parts.length - 2)];
+    if (nextToLastItem === "co" || nextToLastItem === "com") {
+      // Checks to see if a third-level domain is present.
+      // Example: https://www.amazon.co.uk returns "amazon.co.uk"
+      defaultDomainLength = 4;
+    }
+  }
+
+  while (parts.length > (defaultDomainLength - 1)) {
+    // Trim around down to final domain
+    parts.shift();
+  }
+
+  let rootDomain = parts.join(".");
+  return rootDomain;
+};
```

Comment:
Lots of `let`s in here too. Can we include `psl.min.js` in the panel and use that library here too?